### PR TITLE
add ZQL build dependencies to zql/Makefile

### PR DESCRIPTION
**Pull request by nwt**
A more contained alternative to the combination of #283 and #292.

### CODE REVIEW

File: zql/Makefile [hunk 1]
```diff
@@ -9,9 +7,15 @@ run: all
...
+	cd $(deps) && GOBIN=$(deps)/bin go get \
+		github.com/mna/pigeon@v1.0.1-0.20190520151103-9fec3898cef8 \
+		golang.org/x/tools/cmd/goimports@v0.0.0-20200204192400-7124308813f3
+	cpp -DGO -E -P zql.peg | $(deps)/bin/pigeon -o $@
+	$(deps)/bin/goimports -w $@
 
 .PHONY: zql.js
 zql.js:
-	cpp -E -P zql.peg | $(PEGJS) -o $@
+	npm install --global --prefix $(deps) pegjs@0.10.0
```
Comment (aswan) [inline zql/Makefile]: -1 on this, I don't think building should have side effects outside of the source tree...

Comment (aswan) [reply zql/Makefile]: derp, overlooked the --prefix bit

File: zql/Makefile [hunk 2]
```diff
@@ -1,6 +1,4 @@
-GOIMPORTS = goimports
-PEGJS = pegjs
-PIGEON = pigeon
+deps = $(CURDIR)/deps
```
Comment (aswan) [inline zql/Makefile]: CURDIR isn't great -- if I run `make -C zql` from the zq directory this puts the deps in a different place.

Comment (nwt) [reply zql/Makefile]: @aswan: I see this:
```sh
$ pwd
/Users/noah/zq
$ make -C zql | head -1
mkdir -p /Users/noah/zq/zql/deps
```

Do you get different behavior?

Comment (aswan) [reply zql/Makefile]: oops you're right, `-C` is fine, its `make -f zql/Makefile` that doesn't work with CURDIR.  But the latter doesn't work for lots of other reasons anyway and I don't think that's important...

File: zql/Makefile [hunk 3]
```diff
@@ -9,9 +7,15 @@ run: all
 
 .PHONY: zql.go
 zql.go:
-	cpp -DGO -E -P zql.peg | $(PIGEON) -o $@
-	$(GOIMPORTS) -w $@
+	mkdir -p $(deps)
+	echo 'module deps' > $(deps)/go.mod
+	cd $(deps) && GOBIN=$(deps)/bin go get \
+		github.com/mna/pigeon@v1.0.1-0.20190520151103-9fec3898cef8 \
+		golang.org/x/tools/cmd/goimports@v0.0.0-20200204192400-7124308813f3
```
Comment (aswan) [inline zql/Makefile]: Mostly curious, how were these particular versions chosen?

Comment (nwt) [reply zql/Makefile]: They're the most recent versions available that would generate a zql.go identical to what's currently here.

Comment (aswan) [reply zql/Makefile]: Eh, what's current here is really different from what we had been using until #265 was merged a few days ago which abruptly changed the version.

Comment (nwt) [reply zql/Makefile]: Do you think we should switch to whatever was used before? (Chances are good that a change from pigeon v1.0.0 accounts for the difference.)

Comment (aswan) [reply zql/Makefile]: The fact that 1.0.0 is a released version and the thing that we've done most of our testing with does feel better to me, but 🤷‍♂

**Pull request merged**